code/game/objects/items/RSF.dm

```diff
@@ -8,178 +8,201 @@ RSF
+/obj/item/rsf/proc/formRadial(from)
+	var/list/radial_list = list()
+	for(var/meme in from)//We iterate through all of targets entrys
+		var/atom/test = extractObject(from, meme)
+		//We then add their data into the radial menu
+		radial_list[test.name] = image(icon = test.icon, icon_state = test.icon_state)
+		qdel(test)
+	return radial_list
+
+/obj/item/rsf/proc/check_menu(mob/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
 /obj/item/rsf/afterattack(atom/A, mob/user, proximity)
 	. = ..()
 	if(!proximity)
 		return
-	if (!(istype(A, /obj/structure/table) || isfloorturf(A)))
-		return
-
-	if(iscyborg(user))
-		var/mob/living/silicon/robot/R = user
-		if(!R.cell || R.cell.charge < 200)
-			to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
-			icon_state = "rsf_empty"
-			return
-	else if (matter < 1)
-		to_chat(user, "<span class='warning'>\The [src] doesn't have enough matter left.</span>")
-		icon_state = "rsf_empty"
+	if(!is_allowed(A))
 		return
+	if(use_matter(dispense_cost, user))//If we can charge that amount of charge, we do so and return true
+		playsound(src.loc, 'sound/machines/click.ogg', 10, TRUE)
+		var/atom/meme = new to_dispense(get_turf(A))
+		to_chat(user, "<span class='notice'>[action_type] [meme.name]...</span>")
 
-	var/turf/T = get_turf(A)
-	playsound(src.loc, 'sound/machines/click.ogg', 10, TRUE)
-	switch(mode)
-		if(1)
-			to_chat(user, "<span class='notice'>Dispensing Drinking Glass...</span>")
-			new /obj/item/reagent_containers/food/drinks/drinkingglass(T)
-			use_matter(20, user)
-		if(2)
-			to_chat(user, "<span class='notice'>Dispensing Paper Sheet...</span>")
-			new /obj/item/paper(T)
-			use_matter(10, user)
-		if(3)
-			to_chat(user, "<span class='notice'>Dispensing Pen...</span>")
-			new /obj/item/pen(T)
-			use_matter(50, user)
-		if(4)
-			to_chat(user, "<span class='notice'>Dispensing Dice Pack...</span>")
-			new /obj/item/storage/pill_bottle/dice(T)
-			use_matter(200, user)
-		if(5)
-			to_chat(user, "<span class='notice'>Dispensing Cigarette...</span>")
-			new /obj/item/clothing/mask/cigarette(T)
-			use_matter(10, user)
-
+///A helper proc. checks to see if we can afford the amount of charge that is passed, and if we can docs the charge from our base, and returns TRUE. If we can't we return FALSE
 /obj/item/rsf/proc/use_matter(charge, mob/user)
-	if (iscyborg(user))
+	if(iscyborg(user))
 		var/mob/living/silicon/robot/R = user
-		R.cell.charge -= charge
+		var/end_charge = R.cell.charge - charge
+		if(end_charge >= 0)
+			R.cell.charge = end_charge
+			return TRUE
+		to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
 	else
-		matter--
-		to_chat(user, "<span class='notice'>The RSF now holds [matter]/30 fabrication-units.</span>")
-
-/obj/item/cookiesynth
+		if(matter - 1 >= 0)
```

Comment:
Early return.

Reply:
I thought I was. Do you want me to move the bit at the end into each if/else block or something?